src/Controller/ModulesController.php

```diff
@@ -350,6 +350,26 @@ protected function saveRelated(array $relatedData, string $id): void
             $method = (string)Hash::get($rel, 'method');
             $relation = (string)Hash::get($rel, 'relation');
             $relatedIds = (array)Hash::get($rel, 'relatedIds');
+
+            foreach ($relatedIds as $key => &$data) {
+                if (!empty($data['attributes'])) {
+                    // avoid empty objects
+                    $saveObject = false;
+                    foreach ($data['attributes'] as $key => $field) {
+                        if ($key != 'status' && !empty($field)) {
+                            $saveObject = true;
+                        }
+                    }
+
+                    if ($saveObject == true) {
+                        $object = $this->apiClient->save($data['type'], array_merge([
+                            'id' => $data['id'] ?? null,
+                        ], $data['attributes']));
+                    }
+
+                    $data['id'] = $object['data']['id'];
```

Comment:
minor, suggested by scrutinizer too.
It could be fixed adding `$object = [];` somewhere up in the code and using `$data['id'] = Hash::get($object, 'data.id');` in here.

Reply:
why should we do that?